### PR TITLE
add concept of prompt collection

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: langchain/chains/prompt_collection.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class ConditionalPromptCollection(BasePromptCollection):
+    """Prompt collection that goes through conditionals."""
+
+    default_prompt: BasePromptTemplate
+    conditionals: List[
+        Tuple[Callable[[BaseLanguageModel], bool], BasePromptTemplate]
+    ] = Field(default_factory=list)
+
+    def get_default_prompt(self, llm: BaseLanguageModel) -> BasePromptTemplate:
```
Comment (nfcampos) [inline langchain/chains/prompt_collection.py:26]: Should this just be called `get_prompt`?

**Review by nfcampos — COMMENTED** [review]
Just one comment, LGTM otherwise

**Comment by agola11** [comment]
Can we add an example for `ConversationChain` too? I think one of the benefits of prompt collection is to handle creating the `ChatPromptTemplate` with `MessagesPlaceHolder` already in the right place.

**Comment by hwchase17** [comment]
> Can we add an example for `ConversationChain` too? I think one of the benefits of prompt collection is to handle creating the `ChatPromptTemplate` with `MessagesPlaceHolder` already in the right place.

yes ill add a lot more

**Pull request merged**